ecommerce/api.py

```diff
@@ -483,3 +502,120 @@ def send_receipt_email(application_id):
             EMAIL_RECEIPT,
         )
     )
+
+
+WireTransfer = namedtuple(
+    "WireTransfer",
+    ["id", "learner_email", "amount", "bootcamp_start_date", "bootcamp_name", "row"],
+)
+
+
+def parse_wire_transfer_csv(csv_path):
+    """
+    Read CSV file and convert to WireTransfer objects for further processing
+
+    Args:
+        csv_path (str): Path to the CSV file
+
+    Returns:
+        (list of WireTransfer, list):
+    """
+    with open(csv_path) as csv_file:
+        rows = list(csv.reader(csv_file))
+
+    for col, row in enumerate(rows):
+        if WIRE_TRANSFER_LEARNER_EMAIL in row:
+            header_row = row
+            header_row_index = col
+            break
+    else:
+        raise WireTransferImportException("Unable to find header row")
+
+    header_index_lookup = {}
+    for col, cell in enumerate(header_row):
+        if cell in WIRE_TRANSFER_HEADER_FIELDS:
+            header_index_lookup[cell] = col
+
+    for field in WIRE_TRANSFER_HEADER_FIELDS:
+        if field not in header_index_lookup:
+            raise WireTransferImportException(f"Unable to find column header {field}")
+
+    wire_transfers = [
+        WireTransfer(
+            id=int(row[header_index_lookup[WIRE_TRANSFER_ID]]),
+            learner_email=row[header_index_lookup[WIRE_TRANSFER_LEARNER_EMAIL]],
+            amount=Decimal(row[header_index_lookup[WIRE_TRANSFER_AMOUNT]]),
+            bootcamp_start_date=parse_datetime(
+                row[header_index_lookup[WIRE_TRANSFER_BOOTCAMP_START_DATE]]
+            ),
+            bootcamp_name=row[header_index_lookup[WIRE_TRANSFER_BOOTCAMP_NAME]],
+            row=row,
+        )
+        for row in rows[header_row_index + 1 :]
+    ]
+
+    return wire_transfers, header_row
+
+
+def import_wire_transfer(wire_transfer, header_row):
+    """
+    Import a wire transfer. If the WireTransferReceipt already exists, the information will be updated,
+    otherwise it will be created
+    """
+    user = User.objects.get(email=wire_transfer.learner_email)
+    bootcamp_run = BootcampRun.objects.get(
+        bootcamp__title__iexact=wire_transfer.bootcamp_name,
```

Comment:
Would it be worth checking if the title in the CSV matches either the bootcamp title or bootcamp run title?  Just in case a user accidentally types in the latter instead of the former?

Reply:
That makes sense. If it matches `Bootcamp` or `BootcampRun` it will work fine, and if it matches two separate objects it will error, so that's good.